### DISCUSSION TITLE
e2e testing for projects that use cluster-api

### DISCUSSION THREAD

**Discussion by jmccormick2001** [kubernetes-sigs/cluster-api#6291] [Q&A] [answered]
Hi, do you have an example of a project that uses cluster-api and also has e2e tests based upon the e2e framework within cluster-api?  thanks.

**Comment by sbueringer** [comment, marked as answer]
I'm only aware of CAPI infrastructure providers who are doing something similar (but that's just me).

Infra providers consume core CAPI and then they deploy themselves, but in theory you can use the same pattern to just consume core CAPI and an infra provider of your choice. 

Here's the config file: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_conf.yaml

**Comment by jmccormick2001** [comment]
many thanks, I'll check it out.

**Comment by fabriziopandini** [comment]
AFAIK there is also the VSphere CNI provider
